### PR TITLE
[Profiler]Adds CUPTI profiler support

### PR DESCRIPTION
**Pull request by Cunxiao2002**
## Summary by CodeRabbit

- New Features
  - Added built-in benchmarking to the GEMM example, printing measured latency.
  - Expanded profiling capabilities with event- and Kineto-based benchmarking, optional per-kernel timings, and adaptive warmup/repeat for more stable results.
  - Added support for quantile statistics, configurable return modes (min/max/mean/median), multi-kernel profiling, and optional output suppression.
  - Enabled trace export for deeper analysis.
  - Extended the public profiling API with new optional parameters to configure the above behaviors.

### CODE REVIEW

**Comment by LeiWang1999** [comment]
would be better if we can combine do_bench_kineto and do_bench_event into a same api `do_bench`.

File: tilelang/profiler/__init__.py [hunk 1]
```diff
@@ -223,6 +223,10 @@ def do_bench(
         n_warmup: int = 1,
         n_repeat: int = 1,
         input_tensors: List[torch.Tensor] = None,
+        backend: Literal["event", "cupti"] = "event",
+        grad_to_none: Optional[List[torch.Tensor]] = None,
```
Comment (LeiWang1999) [inline tilelang/profiler/__init__.py]: do we really need parameter `grad_to_none`?

File: tilelang/profiler/bench.py [hunk 2]
```diff
@@ -67,34 +117,74 @@ def do_bench(
...
+        )
+        if quantiles is not None:
+            ret = torch.quantile(times, torch.tensor(quantiles, dtype=torch.float)).tolist()
+            if len(ret) == 1:
+                ret = ret[0]
+            return ret
+        return getattr(torch, return_mode)(times).item()
+
+    elif backend == "cupti":
+        using_nsys = int(os.environ.get('DG_NSYS_PROFILING', 0))
```
Comment (LeiWang1999) [inline tilelang/profiler/bench.py]: DG means `DeepGEMM`, would be better to remove this env and always with using_nsys = False.

File: tilelang/profiler/bench.py [hunk 3]
```diff
@@ -2,6 +2,45 @@
 
 import torch
 from typing import Callable, List, Literal, Optional, Union
+import os
+import sys
+
+
+# from https://github.com/deepseek-ai/DeepGEMM/blob/main/deep_gemm/testing/bench.py
+class suppress_stdout_stderr:
```
Comment (LeiWang1999) [inline tilelang/profiler/bench.py:10]: would be better to removed

**Comment by Cunxiao2002** [comment]
some benchmarks for two profiler
1. example/gemm/example_gemm.py
- event Latency: 0.027775999158620834ms
- cupti Latency: 0.023777000000000045ms
2. example/flash_attention/example_mha_fwd_bhsd.py
- event Latency: 0.01020800042897462845ms
- cupti Latency: 0.00630400000000008731ms
3. example/grouped_gemm/example_grouped_gemm_fwd.py
- event Latency: 0.4819520115852356ms
- cupti Latency: 0.4754229999999989ms

**Review by LeiWang1999 — APPROVED** [review]
